node/modules/graphsync.go

```diff
@@ -1,26 +1,39 @@
 package modules
 
 import (
+	"github.com/filecoin-project/lotus/node/config"
 	"github.com/filecoin-project/lotus/node/modules/dtypes"
 	"github.com/filecoin-project/lotus/node/modules/helpers"
+	"github.com/filecoin-project/lotus/node/repo"
 	"github.com/ipfs/go-graphsync"
 	graphsyncimpl "github.com/ipfs/go-graphsync/impl"
 	gsnet "github.com/ipfs/go-graphsync/network"
 	"github.com/ipfs/go-graphsync/storeutil"
 	"github.com/libp2p/go-libp2p-core/host"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"go.uber.org/fx"
+	"golang.org/x/xerrors"
 )
 
 // Graphsync creates a graphsync instance from the given loader and storer
-func Graphsync(mctx helpers.MetricsCtx, lc fx.Lifecycle, clientBs dtypes.ClientBlockstore, chainBs dtypes.ChainBlockstore, h host.Host) (dtypes.Graphsync, error) {
+func Graphsync(mctx helpers.MetricsCtx, lc fx.Lifecycle, r repo.LockedRepo, clientBs dtypes.ClientBlockstore, chainBs dtypes.ChainBlockstore, h host.Host) (dtypes.Graphsync, error) {
 	graphsyncNetwork := gsnet.NewFromLibp2pHost(h)
 	loader := storeutil.LoaderForBlockstore(clientBs)
 	storer := storeutil.StorerForBlockstore(clientBs)
-	gs := graphsyncimpl.New(helpers.LifecycleCtx(mctx, lc), graphsyncNetwork, loader, storer, graphsyncimpl.RejectAllRequestsByDefault())
+	raw, err := r.Config()
+	if err != nil {
+		return nil, err
+	}
+
+	cfg, ok := raw.(*config.FullNode)
```

Comment:
DI modules shouldn't touch the config, it should be handled in node/builder.go:ConfigFullNode (modules.StorageMiner is a good example)

(I'll push a fix)